apps/mobile/src/features/activity/activity-list.tsx

```diff
@@ -1,48 +1,130 @@
-import { useState } from 'react';
+import { useMemo, useState } from 'react';
 
-import { useRefreshHandler } from '@/components/page/page.layout';
+import { Balance } from '@/components/balance/balance';
+import { useBrowser } from '@/core/browser-provider';
+import { useRefreshHandler } from '@/hooks/use-refresh-handler';
 import { ViewMode } from '@/shared/types';
+import { useSettings } from '@/store/settings/settings';
 import { FlashList } from '@shopify/flash-list';
+import { ResponsiveValue } from '@shopify/restyle';
 
-import { Activity, OnChainActivity, OnChainActivityTypes } from '@leather.io/models';
-import { Box } from '@leather.io/ui/native';
+import {
+  Activity,
+  FungibleCryptoAsset,
+  OnChainActivity,
+  OnChainActivityTypes,
+} from '@leather.io/models';
+import { ActivityAvatarIcon, Box, Theme } from '@leather.io/ui/native';
 
 import { ActivityCard } from './activity-card';
 import { ActivityEmpty } from './activity-empty';
 import { ActivityListItem } from './activity-list-item';
+import { formatActivityItem } from './utils/format-activity';
+import { makeActivityLink } from './utils/make-activity-link';
 
 interface ActivityListProps {
   activity: Activity[];
   mode?: ViewMode;
 }
 
 export function ActivityList({ activity, mode = 'full' }: ActivityListProps) {
+  const { networkPreference } = useSettings();
+  const { linkingRef } = useBrowser();
   const { refreshing, onRefresh } = useRefreshHandler();
-  const [renderLimit, setRenderLimit] = useState(10);
-  const filteredActivities = activity
-    .slice(0, renderLimit)
-    .filter(activity => activity.type in OnChainActivityTypes) as OnChainActivity[];
+  const initialRenderLimit = 10;
+  const [renderLimit, setRenderLimit] = useState(initialRenderLimit);
+
+  const activityItems = useMemo(
```

Comment:
I added `useMemo` here for now but [I am looking into a better performance improvement](https://github.com/leather-io/mono/pull/1392) using React Compiler so will likely revert this.

I am using `formatActivityItem` to preformat the data as much as I can to try and simplify the `renderItem` component used by `FlashList` - (`ActivityListItem`). 

In their docs they recommend this being as simple a component as possible. I refactored it down as much as I could but the data we are showing there still needs some other subcomponents:
- `ActivityAvatarIcon`
- `Balance` 

We can look into further improving those but I think this performance improvement is good enough for now